docs/src/architecture/data-model.md

```diff
@@ -0,0 +1,143 @@
+  * Each **Capability** object is of one the of supported **Capanbilities Types** (e.g. `tedge_config`, `tedge_logging`, `tedge_software`). More details see section below [Capability Types](#capability-types).
+
+  * A device **object** can contain several **capability** objects.
+
+## Capability Types
+
+**Capability types** define the contract between a device and a plugin. That contract must be implemented by both parties (device and plugin). **thin-edge** does not interpret the content of any **capability** object, but forwards it to the coresponding plugin, and stores it in the device object in the inventory.<br/>
+NOTE: Some **Capabilities Types** are implemented by built-in thin-edge components (e.g. Software Management). In that case there is no plugin, but the built-in thin-edge component gets the **capability** object fordwarded, and implements that contract.
+
+The structure of **Capability Types** is as below:
+* A **capability type** has a unique name and a set of fields that are expected in the coresponding capability object.
+* Each **capability** object in the inventory is of one of a supported **capability type**.
+
+The following JSON code shows a small example of an object of **capability type** `tedge_config` in a device object:
+         
+          {
+              "name": "child-device 1",
+              "type": "thin-edge.io-child",
+              "capabilities": {
+                  // find below a capability object of type 'tedge_config'
+                  "tedge_config": {
+                      files: [ "foo.conf", "bar.conf" ]
+                  }
+              }
+          }
+
+**thin-edge** as well as installed plugins use those **capability** objects to provide the right **capabilities** to the right devices.
+Furthermore all needed information for the certain **capability** are contained in the **capability** object.
+
+Example:
+
+* The configuration management plugin provides the capability `tedge_config`. 
+* The configuration management plugin recognizes (or gets a request for) a **capability** object for the corresponding device, and processes configuration management for that devices.
+  
+  **Open Topic:** To be decided how the **capability** object comes to the plugin. 
+  * Shall the plugin listen (subscribe) to the inventory, and by itself process any incoming/changed **capability** object?
+  * Better to have an explicit request/response from some inventory owner (e.g. the tegde_agent) to the plugin, 
+    to register a new **capability** object for a certain device?
+* Each capability object `tedge_config` contains the field `files`, that tells the plugin the list of config-file the corresponding device provides.
+
+**thin-edge** has a set of pre-defined **capability types**. 
+Each plugin can define plugin-specific capability types, or use one of the pre-defined capability type.
+For details about all [Pre-defined Capability Types](#pre-defined-capability-types) see section below.
```

Comment:
Here you might wanna mention why we've created such "pre-defined" types. In my view, it establishes standard a contract that plugin developers and device developers can work on independently, without having to agree on a custom contract between them. This could enable the development of cloud-agnostic plugins.